types/lockfile-info/lockfile-info-tests.ts

```diff
@@ -0,0 +1,26 @@
+import lockfileInfo = require('lockfile-info');
+import * as assert from "power-assert";
+
+assert.equal(typeof lockfileInfo, 'function');
+
+const p = lockfileInfo();
+
+assert.equal(Promise.resolve(p), p, 'returhns a Promise');
+
+p.then(({
+    hasPackageJSON,
+    hasNodeModulesDir,
+    hasLockfile,
+    hasPackageLock,
+    hasShrinkwrap,
+    lockfileVersion,
+}) => {
+    assert.equal(typeof hasPackageJSON, 'boolean');
+    assert.equal(typeof hasNodeModulesDir, 'boolean');
+    assert.equal(typeof hasLockfile, 'boolean');
+    assert.equal(typeof hasPackageLock, 'boolean');
+    assert.equal(typeof hasShrinkwrap, 'boolean');
+    assert.equal(typeof lockfileVersion, 'number');
+
+    assert.ok(lockfileVersion === 1 || lockfileVersion === 2 || lockfileVersion === 3 || isNaN(lockfileVersion));
+});
```

Comment:
```suggestion
lockfileInfo; // $ExpectType (cwd?: string | undefined) => Promise<LockfileInfo>

const p = lockfileInfo();

Promise.resolve(p); // $ExpectType Promise<LockfileInfo>

p.then(({
    hasPackageJSON,
    hasNodeModulesDir,
    hasLockfile,
    hasPackageLock,
    hasShrinkwrap,
    lockfileVersion,
}) => {
    hasPackageJSON; // $ExpectType boolean
    hasNodeModulesDir; // $ExpectType boolean
    hasLockfile; // $ExpectType boolean
    hasPackageLock; // $ExpectType boolean
    hasShrinkwrap; // $ExpectType boolean
    lockfileVersion; // $ExpectType number
});
```
DT is built on types assertions using `dtlint`:
https://github.com/DefinitelyTyped/DefinitelyTyped#my-package-teststs

Reply:
The `Promise.resolve(p)` line doesn't do anything tho, unless it's asserting that it has the internal slots of a Promise by confirming that it has the same identity as `p`. How would I capture that?

Reply:
I think in a simplest way as it can be:
```ts
lockfileInfo('.'); // $ExpectType Promise<LockfileInfo>
lockfileInfo(); // $ExpectType Promise<LockfileInfo>
```
I added assertions, without taking care if some constructs are really useful or not,

Reply:
I can do that, but that checks the type, which isn’t the same thing as asserting it’s a real promise.

Reply:
DT is not executing any code, it's pure 'tsc' check, there is nothing better to gain here, unless the same code (form the same file) is run by some other process elsewhere (say your own project pipeline action)

Reply:
Gotcha, thanks! Will update shortly.